acceptance/lib/pxp-agent/bolt_pxp_module_helper.rb

```diff
@@ -0,0 +1,160 @@
+end
+
+# Starts a task.
+# Block is executed on the "response_dataset" object from do_module_action.
+def run_task(broker, targets, task, files, input = {}, metadata = nil, **kwargs, &block)
+  params = { task: task, input: input, files: files }
+  params[:metadata] = metadata if metadata
+  do_module_action(broker, agent_identities(targets), 'task', 'run', params, **kwargs, &block)
+end
+
+# Checks that a non-blocking request finished successfully.
+# Block is executed on the stdout string.
+def ensure_successful(broker, targets, response_dataset, max_retries: 30, query_interval: 1)
+  transaction_ids = response_dataset.map { |data| data['transaction_id'] }
+  agent_identities(targets).zip(transaction_ids).map do |identity, transaction_id|
+    check_non_blocking_response(broker, identity, transaction_id, max_retries, query_interval) do |result|
+      assert_equal('success', result['status'], 'Action was expected to succeed')
+      assert_equal(nil, result['stderr'], 'Successful action expected to have no output on stderr')
+      assert_equal(0, result['exitcode'], 'Successful action expected to have exit code 0')
+      yield result['stdout'] if block_given?
+    end
+  end
+end
+
+# Checks that a non-blocking request finished unsuccessfully without erroring.
+# Block is executed on the stderr string.
+def ensure_failed(broker, targets, response_dataset, max_retries: 30, query_interval: 1)
+  transaction_ids = response_dataset.map { |data| data['transaction_id'] }
+  agent_identities(targets).zip(transaction_ids).map do |identity, transaction_id|
+    check_non_blocking_response(broker, identity, transaction_id, max_retries, query_interval) do |result|
+      assert_equal('failure', result['status'], 'Action was expected to fail')
+      assert_equal(nil, result['stdout'], 'Successful action expected to have no output on stdout')
+      assert_equal(1, result['exitcode'], 'Successful action expected to have exit code 1')
+      yield result['stderr'] if block_given?
+    end
+  end
+end
+
+# Runs a non-blocking task action on targets, and confirms that it was successful.
+# Block is executed on the stdout string.
+def run_successful_task(broker, targets, task, files, input: {}, metadata: nil, **kwargs)
+  run_task(broker, targets, task, files, input, metadata, **kwargs) do |datas|
+    ensure_successful(broker, targets, datas, **kwargs) do |stdout|
+      yield stdout if block_given?
+    end
+  end
+end
+
+# Runs a non-blocking task action on targets, and confirms that it failed but did not error.
+# Block is executed on the stderr string.
+def run_failed_task(broker, targets, task, files, input: {}, metadata: nil, **kwargs)
+  run_task(broker, targets, task, files, input, metadata) do |datas|
+    ensure_failed(broker, targets, datas, **kwargs) do |stderr|
+      yield stderr if block_given?
+    end
+  end
+end
+
+# Starts a non-blocking task on targets, and confirms that it produced an RPC error
```

Comment:
This should say "blocking", not "non-blocking" -- I'll fix it once the Jenkins tests finish.